src/common/include/gudhi/graph_simplicial_complex.h

```diff
@@ -30,40 +32,133 @@ struct vertex_filtration_t {
   typedef boost::vertex_property_tag kind;
 };
 
-/** \brief Proximity_graph contains the vertices and edges with their filtration values in order to store the result
- * of `Gudhi::compute_proximity_graph` function.
+/** \brief Filtered_edges_container contains the edges with their filtration values, sorted by ascending filtration values,
+ * in order to store the result of `Gudhi::compute_edge_graph` function.
  *
- * \tparam SimplicialComplexForProximityGraph furnishes `Filtration_value` type definition.
+ * \tparam SimplicialComplexForFilteredEdges furnishes `Filtration_value` and `Vertex_handle` types definitions.
  *
  */
-template <typename SimplicialComplexForProximityGraph>
-using Proximity_graph = typename boost::adjacency_list < boost::vecS, boost::vecS, boost::directedS
-, boost::property < vertex_filtration_t, typename SimplicialComplexForProximityGraph::Filtration_value >
-, boost::property < edge_filtration_t, typename SimplicialComplexForProximityGraph::Filtration_value >>;
+template <typename SimplicialComplexForFilteredEdges>
+class Filtered_edges_container {
+ public:
+  /** \brief Type of a `Gudhi::Filtered_edges_container` data structure element.
+   */
+  using Filtered_edge = std::tuple<typename SimplicialComplexForFilteredEdges::Filtration_value,
+                                   typename SimplicialComplexForFilteredEdges::Vertex_handle,
+                                   typename SimplicialComplexForFilteredEdges::Vertex_handle>;
+  /** \brief Type of the `Gudhi::Filtered_edges_container` data structure.
+   */
+  using Filtered_edge_set = std::vector<Filtered_edge>;
+
+  using Filtered_edge_set_iterator = typename Filtered_edge_set::const_iterator;
+  /** \brief Range over the `Gudhi::Filtered_edges_container` data structure.
+   */
+  using Filtered_edge_range = boost::iterator_range<Filtered_edge_set_iterator>;
```

Comment:
Why do we need this when we have Filtered_edge_set which is already a range?

Reply:
It looks like there actually was a good reason for Filtered_edge_range, it acts as a view (std::span in C++20) so you can return a cheap object that references a sub-range of edges_, while returning a vector forces a copy, and we cannot return a vector by reference since we only want a subrange.